packages/medusa/src/services/claim.ts

```diff
@@ -204,6 +209,102 @@ export default class ClaimService extends TransactionBaseService {
     )
   }
 
+  protected async validateCreateClaimInput(
+    data: CreateClaimInput
+  ): Promise<void> {
+    const lineItemServiceTx = this.lineItemService_.withTransaction(
+      this.manager_
+    )
+
+    const { type, claim_items, additional_items, refund_amount } = data
+
+    for (const item of claim_items) {
+      const line = await lineItemServiceTx.retrieve(item.item_id, {
+        relations: ["order", "swap", "claim_order", "tax_lines"],
+      })
+
+      if (
+        line.order?.canceled_at ||
+        line.swap?.canceled_at ||
+        line.claim_order?.canceled_at
+      ) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Cannot create a claim on a canceled item.`
+        )
+      }
+    }
+
+    if (type !== ClaimType.REFUND && type !== ClaimType.REPLACE) {
```

Comment:
Suggestion: move those checks before the loop so that if they are not valid we don't query that much